File: TrabajoEnClases/Trabajo5/arbol/Principal.java
package arbol;

public class Principal {
    public static void main(String[] args){
        ArbolBinario arbol = new ArbolBinario();
        Nodo n1,n2,n3,n4,n5,n6,n7;

        n1 = arbol.nuevoArbol(null, "M", null);
        n2 = arbol.nuevoArbol(null, "R", null);
        n3 = arbol.nuevoArbol(n1, "E", n2);

        n4 = arbol.nuevoArbol(null, "A", null);
        n5 = arbol.nuevoArbol(null, "B", null);
        n6 = arbol.nuevoArbol(n4, "MJ", n5);

        n7 = arbol.nuevoArbol(n3, "L", n6);

        ArbolBinario arbola = new ArbolBinario(n7);

        System.out.println(n7.getValorNodo());
        System.out.println(n7.getSubArbolIzq().getValorNodo()+"\t"+n7.getSubArbolDer().getValorNodo());
        System.out.println();
        Imprimir.printNodo(n7);

    }

}

File: tarea7/ruby/Principal.rb
require 'java'
require_relative "ArbolBinario"
java_import 'ruby.Imprimir'
require_relative "Nodo"
require_relative 'Recorrido'

arbol = ArbolBinario.new

n1 = arbol.nuevo_arbol(nil, "s", nil)
n2 = arbol.nuevo_arbol(nil, "a", nil)
n3 = arbol.nuevo_arbol(n1, "e", n2)

n4 = arbol.nuevo_arbol(nil, "f", nil)
n5 = arbol.nuevo_arbol(nil, "g", nil)
n6 = arbol.nuevo_arbol(n4, "h", n5)

n7 = arbol.nuevo_arbol(n3, "p", n6)

puts "Arbol"
Imprimir.printNodo(n7)

preorden(n7)
puts "Preorden"

inorden(n7)
puts "Inorden"

postorden(n7)
puts "Postorden"
